x-pack/plugins/features/common/feature.ts

```diff
@@ -4,16 +4,16 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { FeatureKibanaPrivileges, FeatureKibanaPrivilegesSet } from './feature_kibana_privileges';
+import { RecursiveReadonly } from '@kbn/utility-types';
+import { FeatureKibanaPrivileges } from './feature_kibana_privileges';
+import { ISubFeature, SubFeature } from './sub_feature';
 
 /**
  * Interface for registering a feature.
  * Feature registration allows plugins to hide their applications with spaces,
  * and secure access when configured for security.
  */
-export interface Feature<
-  TPrivileges extends Partial<FeatureKibanaPrivilegesSet> = FeatureKibanaPrivilegesSet
-> {
+export interface IFeature {
```

Comment:
If I understand correctly, we're only using `IFeature` to enable `FeatureRegistry::registerFeature`. What do you think about renaming this to `FeatureConfig` or `RawFeature`? It took me a few minutes to understand when `IFeature` was supposed to be used instead of `Feature` as we've used interfaces before to differentiate between the public contract and the actual implementation. 

This logic would extend to the naming of `ISubFeature`/`SubFeature`, `ISubFeaturePrivilegeGroup` and `ISubFeaturePrivilege`.

Reply:
It's funny you mention this -- I had implementations that used all of your suggested names in the past, and (for reasons lost to vacation) ended up at `IFeature` in the end.

I'm happy to revisit this though. Do you have a preference between `FeatureConfig` or `RawFeature`?

Reply:
I'm gravitating to `FeatureConfig`